treeherder/perf/autoclassify.py

```diff
@@ -0,0 +1,84 @@
+import json
+from datetime import timedelta
+
+from treeherder.etl.common import make_request
+from treeherder.model.derived import JobsModel
+from treeherder.perf.models import PerformanceAlertSummary
+from treeherder.webapp.api.utils import UrlQueryFilter
+
+
+def get_result_set(repository, result_set_id):
+    filter = UrlQueryFilter({"id": result_set_id})
+    full = filter.pop("full", "true").lower() == "true"
+    with JobsModel(repository) as jm:
+        result = jm.get_result_set_list(0, 1, full, filter.conditions)[0]
+    return result
+
+
+def is_downstream(summary):
+    """Returns a Bool of whether or not the summary given is upstream"""
+    for i in range(int(summary.result_set_id), int(summary.prev_result_set_id), -1):
+        try:
+            result_set = get_result_set(summary.repository.name, i)
+        except IndexError:
+            return False
+        if "merge" in result_set["comments"].lower():
+            return True
+        for revision in result_set["revisions"]:
+            if "merge" in revision["comments"].lower():
+                return True
+    return False
+
+
+def get_revision_from_hg(summary):
+    """Get complete revision list from hg.mozilla.org"""
+    revisions = []
+    for i in range(int(summary.result_set_id), int(summary.prev_result_set_id), -1):
+        result_changeset = get_result_set(summary.repository.name, i)["revision"]
+        r = json.loads(make_request("https://hg.mozilla.org/integration/{}/json-pushes/?full=1&version=2&changeset={}".format(summary.repository.name, result_changeset)).text)
+        for push in r["pushes"].keys():
+            revisions.extend([changeset["node"] for changeset in r["pushes"][push]["changesets"]])
+    return revisions
+
+
+def classify_downstream(downstream, upstream_ids):
```

Comment:
Let's call this `classify_possible_downstream`. Sometimes (very rarely) merges do introduce regressions on their own.